Source/SwiftLintFramework/Reporters/CheckstyleReporter.swift

```diff
@@ -25,11 +25,11 @@ public struct CheckstyleReporter: Reporter {
     }
 
     private static func generateForSingleViolation(_ violation: StyleViolation) -> String {
-        let file: String = violation.location.file ?? "<nopath>"
+        let file: String = violation.location.file?.escapedForXml() ?? "<nopath>"
```

Comment:
`"<nopath>"` needs to be escaped: `(violation.location.file ?? "<nopath>").escapedForXml()`